### DISCUSSION TITLE
Restricting one layer from another using incompatible

### DISCUSSION THREAD

**Discussion by ammarns44** [nftchef/art-engine#109] [Help + Q&A]
I tried to restrict one layer from another using the incompatible function but it doesn't work. Is there any fix or is my syntax just wrong. 
Appreciate your help!
![image](https://user-images.githubusercontent.com/97749168/158042316-44945cb7-df92-4517-b97b-858345536258.png)

**Comment by nftchef** [comment]
It looks like you're trying to add an incompatible configuration for a root folder (`Background`) which will not work. It will only work on individual traits–single files or weighted (optional) folders.

If you can explain a little bit more about what you're trying to achieve and share the layersOrder, I can help try and get the desired result

**Reply by ammarns44** [reply]
Thank you for the reply! To my understanding, I can only restrict one trait of background with one trait of effect and not restrict the whole layer? I will be attaching my layers order so you can see.
![image](https://user-images.githubusercontent.com/97749168/158092804-ac263b3c-69f7-4c8e-99c9-06ad6d9430d5.png)

**Reply by ammarns44** [reply]
I want to try and allow some backgrounds to have effects such as sparkle, glow etc. and also want to generate art with a normal background with no effects.

**Reply by nftchef** [reply]
you can restrict 1 background from using any number of `Effect` files, but not the whole folder entirely.
if you want Effect to be _Optional_ You can do either or both of the following
- a. Add a NONE#50.png, where the weight of NONE will control how often no effect layer is used
- b. Nest the effects layer(s) inside subfolders in the `Background` folder, example:
<img width="217" alt="image" src="https://user-images.githubusercontent.com/91582112/158186054-7ef558e0-e4fd-4648-ba8d-94a36d06d4c5.png">
Using `z10,` on the `Effect` folder here will layer it on top of all the other items to maintain your original stacking order (assuming z10 is the highest z-index in your folders/files)
